src/proxy-ui-api/frontend/src/views/AddSubsystem/AddSubsystem.vue

```diff
@@ -1,83 +1,317 @@
+      this.$router.replace({ name: RouteName.Clients });
+    },
+    done(): void {
+      this.$store
+        .dispatch('addSubsystem', {
+          memberName: this.client.member_name,
+          memberClass: this.client.member_class,
+          memberCode: this.client.member_code,
+          subsystemCode: this.subsystemCode,
+        })
+        .then(
+          (response) => {
+            this.disableDone = false;
+
+            if (this.registerChecked) {
+              this.registerSubsystem();
+            } else {
+              this.exitView();
+            }
+          },
+          (error) => {
+            this.$store.dispatch('showError', error);
+          },
+        );
+    },
+
+    registerSubsystem(): void {
+      this.$store
+        .dispatch('registerClient', {
+          instanceId: this.client.instance_id,
+          memberClass: this.client.member_class,
+          memberCode: this.client.member_code,
+          subsystemCode: this.subsystemCode,
+        })
+        .then(
+          (response) => {
+            this.disableDone = false;
+            this.exitView();
+          },
+          (error) => {
+            this.$store.dispatch('showError', error);
+          },
+        );
     },
-    cancel() {
+
+    clientDetailsReady(): void {
+      this.$store.dispatch('showSuccess', 'subsystem_code');
+    },
+
+    exitView(): void {
+      this.$store.dispatch('resetState');
+      this.$store.dispatch('resetAddClientState');
       this.$router.replace({ name: RouteName.Clients });
     },
+    saveSelectedClient(selectedMember: Client): void {
+      this.$store.dispatch('setSelectedMember', selectedMember).then(
+        (response) => {
+          this.$store.dispatch('fetchReservedSubsystems', selectedMember);
+        },
+        (error) => {
+          this.$store.dispatch('showError', error);
+        },
+      );
+      this.showSelectClient = false;
+    },
+    fetchData(): void {
+      // Fetch "parent" client from backend
+      this.$store.dispatch('fetchClient', this.clientId).then(
+        (response) => {
+          this.$store.dispatch('fetchSelectableForSubsystem', this.client);
+        },
+        (error) => {
+          this.$store.dispatch('showError', error);
+        },
+      );
+    },
+  },
+
+  created() {
+    this.fetchData();
   },
 });
 </script>
 
 <style lang="scss" scoped>
-.input-title {
-  max-width: 160px;
-  text-align: left;
+@import '../../assets/colors';
+@import '../../assets/shared';
+@import '../../assets/wizards';
+
+.view-wrap {
+  width: 100%;
+  max-width: 1000px;
+  margin: 10px;
 }
 
-.subs-input {
-  text-align: left;
+.view-title {
+  width: 100%;
+  max-width: 100%;
+  margin-bottom: 30px;
 }
 
-.input-name {
-  padding-top: 0;
-  margin-top: 0;
+.info-block {
+  display: flex;
+  flex-direction: row;
+  margin-bottom: 40px;
+
+  .action-block {
+    margin-top: 30px;
+    margin-left: auto;
+    margin-right: 0px;
+  }
```

Comment:
Perhaps refactor styles here so that they're consistent in using hierarchial style definitions. .view-wrap contains all the other class style definitions in this comment rows. Currently only info-block and action-block hierarchy is defined.

Reply:
Fixed